src/algebra/big_operators/basic.lean

```diff
@@ -1012,6 +1012,21 @@ end prod_eq_zero
 
 end finset
 
+namespace list
+
+@[to_additive, simp] lemma prod_to_finset {M : Type*} [decidable_eq α] [comm_monoid M]
+  (f : α → M) {l : list α} (hl : l.nodup) : l.to_finset.prod f = (l.map f).prod :=
```

Comment:
Does this work as a simp lemma, given it needs `hl`?

Reply:
Hmm, good point.